### ISSUE TITLE
FileUrl test in checkDocServiceUrl() shouldn't be called on plugin configuration

### ISSUE THREAD

**Issue by ywarnier** [ONLYOFFICE/onlyoffice-chamilo#40]
Another error appears on saving the plugin details:
`Error when trying to connect(cURL error 3: URL using bad/illegal format or missing URL (see https://curl.haxx.se/libcurl/c/libcurl-errors.html))(Version 8.1.3.4)`

This is due to the system checking, during the plugin configuration, for a link to a *file* instead of just testing the connection.

In vendor/onlyoffice/docs-integration-sdk/src/Service/Request/RequestService.php, line ~430:
```
        try {
            $fileUrl = $this->getFileUrlForConvert();

            if (!empty($fileUrl)) {
                if (!empty($this->settingsManager->getStorageUrl())) {
                    $fileUrl = str_replace(
                        $this->settingsManager->getServerUrl(),
                        $this->settingsManager->getStorageUrl(),
                        $fileUrl
                    );
                }
                $convertedFileUri = $this->getConvertedUri($fileUrl, "docx", "docx", "check_" . rand());
            }
        } catch (\Exception $e) {
            return [$e->getMessage(), $version];
        }

        try {
            $this->request($convertedFileUri);
        } catch (\Exception $e) {
            return [$e->getMessage(), $version];
        }
```
This whole condition should *not* be tested when configuring the plugin, because there is no file, so `$fileUrl` remains empty and the error shown appears.

![Capture d’écran du 2024-10-16 14-32-56](https://github.com/user-attachments/assets/52242ee5-875b-4c05-a631-da2b37152f90)

**Comment by ywarnier** [comment]
In my view, the call to checkDocServiceUrl() should come with a setting to check (or not) the existence of a link to a file.
This would allow the plugin configuration section to skip testing the file, while the rest of the plugin continues testing it.

**Comment by rivexe** [comment]
@ywarnier Hello! Thank you for your attention! Please tell me the following:
1. When testing conversion, we create a `convert.docx` file in the plugin directory and pass a link to it. We didn't see any problems with this in our test builds. Perhaps this behavior is unacceptable in general? If so, we can remove this check for the plugin. 
2. Could you share your php configuration?

Thank you!

**Comment by ywarnier** [comment]
Ah, sure. The plugin/ directory should be non-writeable by the web server, for security reasons, so in my case it isn't writeable. Please use api_get_path(SYS_ARCHIVE_PATH) to get the path to the temp (app/cache/) directory (if that works with the plugin logic). You can create files and/or subdirectories in there all you want.

**Comment by rivexe** [comment]
@ywarnier One more question. Do we mean that the `app/cache` directory will be accessible from the network? I mean, will our editor server be able to access the Chamilo instance using the URL generated by `api_get_path(WEB_ARCHIVE_PATH)`...? Thank you.

**Comment by ywarnier** [comment]
Can't check in the next few hours, but I think it is. Would have to check the .htaccess file.

**Comment by ywarnier** [comment]
OK, I just checked and you *cannot* access `app/cache/` directly without permissions.
However, `app/upload/plugins/` should also be writeable (but blocked from executing any PHP code to prevent hacks), and that one you can access directly. So you could do something like:
```
$basePath = api_get_path(SYS_UPLOAD_PATH).'plugins/';
if (!is_dir($basePath.'onlyoffice')) {
    api_create_protected_dir('onlyoffice',$basePath);
}
file_put_contents($basePath.'onlyoffice/'.$fileName, $fileContents);
```
And then you can access the public path of that file with `api_get_path(WEB_UPLOAD_PATH).'plugins/onlyoffice/'.$fileName`.
Please only use that to input a preset file with a known filename. Don't let users choose as this would require much stricter filtering of file name, file type and file contents.

**Comment by rivexe** [comment]
@ywarnier  Thank you for your quick feedback and active participation! I think we can not create a temporary file to check the conversion at all, but simply give the file stream to the editor along a specific route. As part of the plugin, we have templates for empty documents, and we will take the stream from such a template. Edits are pending, I think the problem will be resolved.

**Comment by ywarnier** [comment]
Sounds great !

**Comment by rivexe** [comment]
@ywarnier Can you check the 'develop' branch now? We fixed the conversion check when saving settings and the problem from your [other issue](https://github.com/ONLYOFFICE/onlyoffice-chamilo/issues/38). It works on our test instances.

**Comment by rivexe** [comment]
If you will test the work of the editors themselves on documents, I'll clarify, just in case, that during building we get git submodules in addition to composer dependencies.

**Comment by LinneyS** [comment]
@ywarnier 
Please check if there are any more issues?
If everything is ok, we will release fixes.
Also, please share your php config for the environment so that we can see similar issues. Thanks

**Comment by ywarnier** [comment]
This is fixed as well.
*however* could you please wait about an hour to release a new version ? I'm adding a feature in Chamilo to open office documents from the learning path, but some changes need to be applied to the plugin in order to do that and I would be really happy to have them in your version 1.5 and then include them immediately in Chamilo 1.11.28. I will open a new ticket about this so we can close this one here.

**Comment by LinneyS** [comment]
Released as [v1.4.1](https://github.com/ONLYOFFICE/onlyoffice-chamilo/releases/tag/v1.4.1)
